eng/Packages.Data.props

```diff
@@ -120,13 +120,13 @@
 
   <ItemGroup Condition="$(MSBuildProjectName.StartsWith('Azure.Monitor.OpenTelemetry'))">
     <!-- OpenTelemetry dependency approved for Azure.Monitor.OpenTelemetry.Exporter package only -->
-    <PackageReference Update="OpenTelemetry" Version="1.4.0" />
-    <PackageReference Update="OpenTelemetry.Exporter.InMemory" Version="1.4.0" />
+    <PackageReference Update="OpenTelemetry" Version="[1.5.0, 1.6.0)" />
```

Comment:
Accepts version 1.5.0 up to 1.5.x on update, but not 1.6 and higher.

The Instrumentation Libraries are implementing breaking changes wrt Http Semantic Conventions. We want to use this to lock the Exporter to the v1.5.0 milestone.

Reply:
I am not sure why this is needed.

Reply:
I recommend to keep the scope of this PR to just upgrade the OpenTelemetry SDK without introducing additional dependencies. 

You could propose this change in a follow up PR.

Reply:
reverted this change